File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1
        Maxwater=0
        while(i<j):
            area=min(height[i],height[j])*(j - i)
            Maxwater=max(Maxwater,area)
            if(height[i]>=height[j]):
                j-=1
            else:
                i+=1
        return Maxwater
        

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0;
        int j=height.size() - 1;
        int Maxwater=0;
        while(i<j)
        {
            int area= std::min(height[i],height[j])*(j - i);
            Maxwater=std::max(Maxwater,area);
            if(height[i]>=height[j])
            {
                j-=1;
            }
            else
            {
                i+=1;
            }
            
        }
        return Maxwater;
        
    }
};